### PR TITLE
Update bookmarklet with magic

### PR DESCRIPTION
**Pull request by jfexyz**
#### What's this PR do?

Updates the template bookmarklet to download the template from Github on-demand. Auto-updating #FTW! Also adds the ability to overwrite a current PR body, with confirmation!
#### Where should the reviewer start?

Look at all the things.
#### How should this be manually tested?
- [x] Add the bookmarklet to your bookmarks bar.
- [x] Edit the new bookmarklet, changing `https://github.com/attn/pull-request-template/blob/master/pull-request-template.md` to `https://github.com/joshuajabbour/pull-request-template/blob/update-bookmarklet/pull-request-template.md` (required for testing only).
- [x] Create a new pull request, click the bookmarklet, and boom.
  - [x] Delete the textarea content, add some dummy text, and click the bookmarklet again.
  - [x] Confirm or deny replacement of current text. Make sure it works.
- [x] Edit old pull request, click again. Confirm or deny.
- [x] Repeat until you are bored. Test one more time.
#### Any background context you want to provide?
- I wanted to be able to add the template into old PRs.
- Also, I want to change it (PR forthcoming) and have everyone use the correct template.
- Finally, I like being clever.
#### What are the relevant tickets?

Wut tix?
#### Screenshots (if appropriate)

Haha.
#### Questions:
- [ ] Do you like me, yes or no?
#### Checklist:
- [x] Code standard review
- [x] Documentation provided
#### Release Type:
- [ ] Major
- [x] Minor
- [ ] Patch

### CODE REVIEW

**Comment by jfexyz** [comment]
Just FYI: In theory, it would be awesome to use Github's raw content URLs\* to load this, instead of parsing the content out of their HTML. But since they are on a different domain, CORS won't allow that. Hence the parsing...

\* https://raw.githubusercontent.com/joshuajabbour/pull-request-template/update-bookmarklet/pull-request-template.md

**Comment by desmondmorris** [comment]
@joshuajabbour probably wont solve the COR's problemo, but worth a shot - https://rawgit.com/

**Comment by jfexyz** [comment]
Nope, tried it; doesn't work either. Github uses `connect-src self` to declare the domains from which content can be loaded, which are set to a few of theirs only. That is what limits loading, and since neither rawgit.com nor githubusercontent.com are included, you can't load content from there. #security

**Comment by jfexyz** [comment]
:ghost:

File: README.md [hunk 1]
```diff
@@ -1,39 +1,14 @@
-Pull Request Template
-=====================
+attn: Pull Request Template
```
Comment (mmwtsn) [inline README.md]: Company name should be capitalized when written out as per [our brand guidelines](https://sites.google.com/a/attn.com/documentation/home/organization/brand-guidelines). Alternately, we could throw the logo in [like this](https://github.com/attn/automation/blob/master/README.md).

File: README.md [hunk 2]
```diff
@@ -1,39 +1,14 @@
...
+===========================
```
Comment (mmwtsn) [inline README.md]: Why not use a single `#` here? Less to maintain.

Comment (jfexyz) [inline README.md]: Mainly bc I didn't change it from what was there before. :)

File: README.md (hunk 1 above)
Comment (jfexyz) [inline README.md]: Good to know. Been wondering that; almost asked when I wrote this.

Comment (jfexyz) [inline README.md]: (But don't get me started on the discordance between our name and logo. #ohwell)

Comment (mmwtsn) [inline README.md]: Me neither. Just enforcing our standard :)

Comment (mmwtsn) [inline README.md]: I asked about this when I joined and the rationale at the time was capitalizing our name increased its visibility on Facebook which increased traffic to the site. Lowercase letters look better in the logotype, however, so we use both.

Comment (jfexyz) [inline README.md]: Haha, I guarantee you there is no proof for that statement.

Comment (jfexyz) [inline README.md]: And shouldn't we also update the site's `<title>` tag?

Comment (jfexyz) [inline README.md]: See, I said DON'T GET ME STARTED! :stuck_out_tongue:

**Comment by mmwtsn** [comment]
Just a general FYI JavaScript bookmarklets will not work on websites served over HTTPS in Firefox (see [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=866522)). That's not a problem for anyone on this team at time of writing as everyone else is a Chrome user.

Will pick up testing in Chrome ~~in a few~~ tomorrow morning.

**Comment by mmwtsn** [comment]
Everything is working expected except the template itself. The bookmarklet is grabbing the rendered text as opposed to the raw markdown which results in a plain text pull request template:

pull-request

**Comment by jfexyz** [comment]
That's weird, it totally worked in my testing. I'll look into this some more...

**Comment by mmwtsn** [comment]
Closing due to inactivity.

**Pull request closed without merging**